stackslib/src/chainstate/nakamoto/coordinator/mod.rs

```diff
@@ -0,0 +1,720 @@
+    ) -> bool {
+        // timeout so that we handle Ctrl-C a little gracefully
+        let bits = comms.wait_on();
+        if (bits & (CoordinatorEvents::NEW_STACKS_BLOCK as u8)) != 0 {
+            signal_mining_blocked(miner_status.clone());
+            debug!("Received new Nakamoto stacks block notice");
+            match self.handle_new_nakamoto_stacks_block() {
+                Ok(missing_block_opt) => {
+                    if missing_block_opt.is_some() {
+                        debug!(
+                            "Missing affirmed anchor block: {:?}",
+                            &missing_block_opt.as_ref().expect("unreachable")
+                        );
+                    }
+                }
+                Err(e) => {
+                    warn!("Error processing new stacks block: {:?}", e);
+                }
+            }
+
+            signal_mining_ready(miner_status.clone());
+        }
+        if (bits & (CoordinatorEvents::NEW_BURN_BLOCK as u8)) != 0 {
+            signal_mining_blocked(miner_status.clone());
+            debug!("Received new burn block notice");
+            match self.handle_new_nakamoto_burnchain_block() {
+                Ok(missing_block_opt) => {
+                    if missing_block_opt.is_some() {
+                        debug!(
+                            "Missing canonical anchor block {}",
+                            &missing_block_opt.clone().unwrap()
+                        );
+                    }
+                }
+                Err(e) => {
+                    warn!("Error processing new burn block: {:?}", e);
+                }
+            }
+            signal_mining_ready(miner_status.clone());
+        }
+        if (bits & (CoordinatorEvents::STOP as u8)) != 0 {
+            signal_mining_blocked(miner_status.clone());
+            debug!("Received stop notice");
+            return false;
+        }
+
+        return true;
+    }
+
+    /// Handle one or more new Nakamoto Stacks blocks.
+    /// If we process a PoX anchor block, then return its block hash.  This unblocks processing the
+    /// next reward cycle's burnchain blocks.
+    pub fn handle_new_nakamoto_stacks_block(&mut self) -> Result<Option<BlockHeaderHash>, Error> {
+        let canonical_sortition_tip = self.canonical_sortition_tip.clone().expect(
+            "FAIL: processing a new Stacks block, but don't have a canonical sortition tip",
+        );
+
+        loop {
+            // process at most one block per loop pass
+            let sortdb_handle = self
+                .sortition_db
+                .tx_handle_begin(&canonical_sortition_tip)?;
+
+            let mut processed_blocks = NakamotoChainState::process_nakamoto_blocks(
+                &mut self.chain_state_db,
+                sortdb_handle,
+                1,
+                self.dispatcher,
+            )?;
+
+            if processed_blocks.len() == 0 {
+                // out of blocks
+                break;
+            }
```

Comment:
Given that we only process nakamoto blocks 1 at a time, does it makes sense to keep the `process_nakamoto_blocks` function which is just a looping wrapper around `process_next_nakamoto_block`?

Reply:
I can do that.